indico/modules/events/registration/tasks.py

```diff
@@ -52,9 +59,25 @@ def delete_field_data():
             data.data = data.field_data.field.field_impl.default_value
             if data.field_data.field.field_impl.is_file_field:
                 logger.debug('Deleting file: %s', data.filename)
-                try:
-                    data.delete()
-                except Exception as e:
-                    logger.error('Failed to delete file for %r: %s', data, e)
+                _delete_file(data)
             data.field_data.field.is_purged = True
     db.session.commit()
+
+
+@celery.periodic_task(name='registration_retention_period', run_every=crontab(minute='0', hour='3'))
+def delete_registrations():
+    registrations = (Registration.query
+                     .join(RegistrationForm, RegistrationForm.id == Registration.registration_form_id)
+                     .join(Event)
+                     .filter(RegistrationForm.retention_period.isnot(None),
+                             db.cast(Event.end_dt, db.Date) + RegistrationForm.retention_period <= now_utc().date())
+                     .all())
+
+    for reg in registrations:
+        logger.info('Purging registration: %r', reg)
+        for data in reg.data:
+            if data.storage_file_id:
+                _delete_file(data)
```

Comment:
I'd keep a `logger.debug('Deleting file: %s', data.filename)` here as well (or move it inside `_delete_file`)